File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/ControlJoystickData.java
package net.kdt.pojavlaunch.customcontrols;

public class ControlJoystickData extends ControlData {

    /* ジョイスティックが前方に固定可能かどうか */
    public boolean forwardLock = false;
    /*
     * 指の追跡が絶対方式（タッチした位置にジョイスティックが移動）か
     * 相対方式（ジョイスティックは中央に留まる）かどうか
     */
    public boolean absolute = false;
/**
 * コンストラクタ。
 * このクラスの新しいインスタンスを初期化します。
 */
    public ControlJoystickData(){
        super();
    }
/**
 * コンストラクタ。
 * このクラスの新しいインスタンスを初期化します。
 */
    public ControlJoystickData(ControlJoystickData properties) {
        super(properties);
        forwardLock = properties.forwardLock;
        absolute = properties.absolute;
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/ControlJoystickData.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols

/** ジョイスティックコントロールの設定データを保持するクラス。ControlDataを継承し、フォワードロックや絶対追跡の設定を追加します。 */
class ControlJoystickData : ControlData {
    /** フォワードロックが有効かどうか。 */
    var forwardLock = false
    /** 絶対座標追跡が有効かどうか。 */
    var absolute = false

    constructor() : super()

    /** @param properties コピー元のControlJoystickData */
    constructor(properties: ControlJoystickData) : super(properties) {
        forwardLock = properties.forwardLock
        absolute = properties.absolute
    }
}